community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -1461,6 +1461,18 @@ class ReservationDetails:
     policies: List[str] # names (not URLs) of resource policies
     bulk_insert_name: str # name in format suitable for bulk insert (currently identical to user supplied name in long format)
 
+@dataclass(frozen=True)
+class FutureReservation:
+    project: str
+    zone: str
+    name: str
+    active: Optional[bool]
+    specific: Optional[bool]
```

Comment:
Why are those fields , except `active_reservation`, are `Optional`. Under what conditions they will not be set?

Reply:
Changed the parameters now only `active_reservation` will ever be `Optional`